Description: Forward allow_unauthenticated to python-apt
 Forward the allow_unauthenticated flag to python-apt so packages can continue
 to be downloaded without them being authenticated, avoiding breakage with recent
 python-apt uploads.
Author: Julian Andres Klode <juliank@ubuntu.com>
Origin: vendor
Bug-Ubuntu: https://bugs.launchpad.net/ubuntu/+source/python-apt/+bug/1858973
--- a/aptdaemon/worker/aptworker.py
+++ b/aptdaemon/worker/aptworker.py
@@ -1068,7 +1068,12 @@ class AptWorker(BaseWorker):
                                               end=install_range[1])
         with self._frozen_status():
             try:
-                self._cache.commit(fetch_progress, inst_progress)
+                # This was backported as 
+                if "allow_unauthenticated" in apt.Cache.commit.__doc__:
+                    self._cache.commit(fetch_progress, inst_progress,
+                                       allow_unauthenticated=trans.allow_unauthenticated)
+                else:
+                    self._cache.commit(fetch_progress, inst_progress)
             except apt.cache.FetchFailedException as error:
                 raise TransactionFailed(ERROR_PACKAGE_DOWNLOAD_FAILED,
                                         str(error))
